crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -274,16 +281,21 @@ impl FormatRule<Suite, PyFormatContext<'_>> for FormatSuite {
                     // it then counts the lines between the statement and the trailing comment, which is
                     // always 0. This is why it skips any trailing trivia (trivia that's on the same line)
                     // and counts the lines after.
-                    lines_after_ignoring_trivia(offset, source)
+                    lines_after(offset, source)
                 };
 
+                let end = comments
+                    .trailing(preceding)
+                    .last()
+                    .map_or(preceding.end(), |comment| comment.slice().end());
```

Comment:
We now look up the comments of every node twice: Once on line 308 and once here. Would it make sense to cache the comments similar to `preceding` (`preceding_commens`)?